domains/certificates/Query.API/API/Query.API/Projections/CertificatesByOwnerProjection.cs

```diff
@@ -1,119 +1,153 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
-using API.Query.API.ApiModels.Responses;
+using System.Threading;
+using System.Threading.Tasks;
+using API.Query.API.Projections.Views;
 using CertificateEvents;
+using CertificateEvents.Exceptions;
+using Marten;
+using Marten.Events;
 using Marten.Events.Projections;
-using Marten.Schema;
 
 namespace API.Query.API.Projections;
 
-public class CertificatesByOwnerProjection : MultiStreamAggregation<CertificatesByOwnerView, string>
+public class CertificatesByOwnerProjection : IProjection
 {
-    public CertificatesByOwnerProjection()
+    public Task ApplyAsync(IDocumentOperations operations, IReadOnlyList<StreamAction> streams,
+        CancellationToken cancellation)
     {
-        Identity<ProductionCertificateCreated>(e => e.MeteringPointOwner);
-        Identity<ProductionCertificateIssued>(e => e.MeteringPointOwner);
-        Identity<ProductionCertificateRejected>(e => e.MeteringPointOwner);
-        Identity<ProductionCertificateTransferred>(e => e.Source);
-        Identity<ProductionCertificateTransferred>(e => e.Target);
+        Apply(operations, streams);
+        return Task.CompletedTask;
     }
 
-    public void Apply(ProductionCertificateTransferred @event, CertificatesByOwnerView view)
+    public void Apply(IDocumentOperations operations, IReadOnlyList<StreamAction> streams)
     {
-        if (view.Owner == @event.Source)
-        {
-            view.Certificates.Remove(@event.CertificateId);
-        }
-        else
-        {
-            view.Certificates[@event.CertificateId] = new CertificateView
+        var events = streams
+            .SelectMany(x => x.Events)
+            .OrderBy(s => s.Sequence)
+            .Select(s => s.Data);
+
+        var view =
```

Comment:
There is a hidden assumption here that we only get events for a single certificate. If there are events from more than one certificate, then this will (probably) fail. I suggest that we make this hidden assumption clear by throwing an exception in case of more than one unique certificate id